tests/tools_test.py

```diff
@@ -126,3 +136,28 @@ def test_combined_curves():
             ],
         ),
     )
+
+
+@pytest.mark.filterwarnings("ignore:No suitable")
+def test_orso_datasets_tool():
+    def normalized_ioq(filename: Filename[SampleRun]) -> ReflectivityOverQ:
+        return filename
+
+    def orso_dataset(filename: Filename[SampleRun]) -> OrsoIofQDataset:
+        class Reduction:
+            corrections = []  # noqa: RUF012
+
+        return OrsoDataset(
+            Orso({}, Reduction, [], name=f'{filename}.orso'), np.ones((0, 0))
+        )
+
+    workflow = sl.Pipeline(
+        [normalized_ioq, orso_dataset], params={Filename[SampleRun]: 'default'}
+    )
+    datasets = orso_datasets_from_measurements(
+        workflow,
+        [{}, {Filename[SampleRun]: 'special'}],
```

Comment:
You are not really testing what you claimed above was needed: Different sets of parameter for different runs. Yes, you pass `{}` for the first, but since it relies on a default value that is not saying much? For what you are testing here, you could have went with the simpler version I had suggested, simply by passing the filename for each. As it is now, what `orso_dataset_from_measurements` will do for the first run is a mystery for the reader.

Reply:
Seems to be a misunderstanding here. Maybe we should talk about this in person? If you have a simpler and better solution in mind I'll be happy to use that.

> You are not really testing what you claimed above was needed: Different sets of parameter for different runs.

We want to override different parameters in different runs. For example, in some run we might want to override the `SampleRotation` parameter, but in another run we want to use the default (meaning reading it from the file).
The intent with this test is to assert if:
1. If no value was provided for a parameter (in this case `Filename`), is the default value used?
2. If a value was provided, is that value used?